### DISCUSSION TITLE
Retaining the style in .osm.pbf file

### DISCUSSION THREAD

**Discussion by fullstackdude** [systemed/tilemaker#522] [Q&A] [answered]
When I use the Ruby server.rb after using tilemaker to create an output.mbtiles file, the resulting map no longer contains all the styling in my original map (map.osm.pbf). Things like parking areas, playgrounds, etc have disappeared and the color styling is also gone. Is there a way to create the vector tiles so they retain this data? Thank you.

**Comment by systemed** [comment]
Neither vector tiles (as produced by tilemaker) or .osm.pbf files contain any styling information, such as colours or line widths. Styling information is always provided by a separate stylesheet. tilemaker ships with a basic example stylesheet just for demo purposes, but creating a stylesheet that matches your own requirements isn't something in tilemaker's scope.

You might find the quick overview at https://github.com/systemed/tilemaker/blob/master/docs/VECTOR_TILES.md helpful in understanding how it works.

**Reply by fullstackdude** [reply]
Thank you for the reply. I'm a noob to osm but have managed to get a map working using renderd to serve tiles. I do need to learn the details but in the meantime just trying to get the same map working with tilemaker. I will substitute the tilemaker demo style files with the ones I've been using.

I asked two questions - should I post the second question separately? My second issue is when I use tilemaker with my map.osm.pbf file, some attributes are not appearing while others are not getting removed. What I've done with my tile based map, is use osmconvert to extract a map.osm file from my map.osm.pbf file, then use osmfilter map.osm --drop="amenity=place_of_worship =farm" -o=newmap.osm. What I've done with tilemaker, is edit the lua file to remove attributes I don't want and keep the ones I do want, but no matter how I have edited the lua files I get the same result. Have also edited the various config.json files.

**Reply by systemed** [reply]
Have you altered the stylesheet JSON file to also look for these newly written objects/attributes, and style them accordingly (i.e. colours/widths/etc.)?

**Reply by fullstackdude** [reply]
The problem here is the documentation doesn't begin to describe the steps necessary. Does any documentation exist elsewhere that goes into the necessary details of configuring Tilemaker? Thank you.

**Reply by systemed** [reply, marked as answer]
The documentation for tilemaker is just documentation for tilemaker, if you see what I mean. :)

tilemaker just generates vector tiles - it doesn't render them. You will usually render them with MapLibre GL or another vector tile rendering library. You need to look at the documentation for that to find out about the stylesheet language. Here's the documentation you probably want: https://maplibre.org/maplibre-style-spec/

**Reply by fullstackdude** [reply]
Thank you for taking the time to educate a noob. I've also purchased a couple of books to learn more.

**Reply by fullstackdude** [reply]
What threw me off is what appears to be filtering in the tilemaker Lua file, references to all the amenities and layers.
